libraries/Microsoft.Bot.Builder.LanguageGeneration/Templates.cs

```diff
@@ -324,7 +324,7 @@ public IList<object> ExpandTemplate(string templateName, object scope = null, Ev
         public AnalyzerResult AnalyzeTemplate(string templateName)
         {
             CheckErrors();
-            var analyzer = new Analyzer(this);
+            var analyzer = new Analyzer(this, this.LgOptions);
```

Comment:
Can we set the flag here, to avoid PVA or anyone have to do it manually

AnalyzeTemplate(string templateName, throwOnRecursive = ture)
{
this.lgOptions.throwOnRecursive = throwOnRecursive ;
var analyzer = new Analyzer(this, this.LgOptions);
}

Reply:
Thank you for this feedback. Changed to an AnalyzeTemplate parameter